feat: introduce new tenancy agreement messages (#49)
Garaio-REM/remcloud-message-bus-api: letting_context.md

    "state":"validated"
  }
}
```

### Letting.TenancyAgreement.DraftUpdated

The validated tenancy agreement has some changes. If the starting date of the tenancy has been updated, we
include a new key `newStartDate` in the message body and send the old starting date in the `startDate` key. 

| Field                    | Type     | Content / Remarks                                                                                                                     |
|--------------------------| -------- |---------------------------------------------------------------------------------------------------------------------------------------|
| eventType                | `string` | Letting.TenancyAgreement.DraftUpdated                                                                                                 |
| data                     | `hash`   |                                                                                                                                       |
| &nbsp;&nbsp;reference    | `string` | unique tenancy agreement identifier, eg '1234.01.0001.01'                                                                             |

> | &nbsp;&nbsp;startDate    | `string` | The starting date of the tenancy agreement. If starting date of tenancy agreement changes, then here we expose the old starting date. |

Reviewer: ```suggestion
| &nbsp;&nbsp;startDate    | `string` | The starting date of the tenancy agreement. Always sent. If starting date of tenancy agreement changes, this will be the old starting date. |
```

> | &nbsp;&nbsp;newStartDate | `string` | The updated starting date of the tenancy agreement. We expose this only when starting date of tenancy changes                         |

Reviewer: ```suggestion
| &nbsp;&nbsp;newStartDate | `string` | The updated starting date of the tenancy agreement. Sent only when starting date of tenancy changes                         |
```

The tenancy agreement has been cancelled.

| Field                    | Type     | Content / Remarks                                                                                                                     |
|--------------------------| -------- |---------------------------------------------------------------------------------------------------------------------------------------|
| eventType                | `string` | Letting.TenancyAgreement.Terminated                                                                                                 |
| data                     | `hash`   |                                                                                                                                       |
| &nbsp;&nbsp;reference    | `string` | unique tenancy agreement identifier, eg '1234.01.0001.01'                                                                             |
| &nbsp;&nbsp;state        | `string` | The current state of the tenancy agreement. It can be in validated, activated, deactivated, cancelled state                           |

#### Example

```json
{
  "eventType":"Letting.TenancyAgreement.Terminated",
  "data": {

>     "reference":"1234.01.0001.01",

Reviewer: All messages should include a startDate to allow the consumer to identify the correct version of the agreement. Please make sure we send & test it for all newly implemented messages and update the docs accordingly.
